### PR TITLE
Move a helper function to prevent circular dependency

### PR DESCRIPTION
**Pull request by martymichal**
Since the `sanitize_filepart` function does not rely on anything in the `perun.profile` module, it can be safely moved out into a more neutral module.

I encountered the following traceback while playing around with Perun.

```
Traceback (most recent call last):                                                                                                                                                                                                                            
  File "/var/home/marty/.local/bin/perun", line 5, in <module>                                                                                                                                                                                                
    from perun.cli import launch_cli                                                                                                                                                                                                                          
  File "/var/home/marty/Documents/projects/perun/perun/cli.py", line 55, in <module>                                                                                                                                                                          
    import perun.fuzz.factory as fuzz                                                                                                                                                                                                                         
  File "/var/home/marty/Documents/projects/perun/perun/fuzz/factory.py", line 27, in <module>                                                                                                                                                                 
    import perun.fuzz.evaluate.by_perun as evaluate_workloads_by_perun                                                                                                                                                                                        
  File "/var/home/marty/Documents/projects/perun/perun/fuzz/evaluate/by_perun.py", line 8, in <module>                                                                                                                                                        
    import perun.check.factory as check                                                                                                                                                                                                                       
  File "/var/home/marty/Documents/projects/perun/perun/check/factory.py", line 15, in <module>                                                                                                                                                                
    from perun.profile.helpers import ProfileInfo                                                                                                                                                                                                             
  File "/var/home/marty/Documents/projects/perun/perun/profile/helpers.py", line 29, in <module>                                                                                                                                                              
    import perun.logic.store as store                                                                                                                                                                                                                         
  File "/var/home/marty/Documents/projects/perun/perun/logic/store.py", line 22, in <module>                                                                                                                                                                  
    from perun.profile.factory import Profile                                                                                                                                                                                                                 
  File "/var/home/marty/Documents/projects/perun/perun/profile/factory.py", line 25, in <module>                                                                                                                                                              
    from perun.check.general_detection import get_filtered_best_models_of                                                                                                                                                                                     
  File "/var/home/marty/Documents/projects/perun/perun/check/general_detection.py", line 24, in <module>                                                                                                                                                      
    import perun.check.fast_check as fast_check                                                                                                                                                                                                               
  File "/var/home/marty/Documents/projects/perun/perun/check/fast_check.py", line 13, in <module>                                                                                                                                                             
    import perun.logic.runner as runner                                                                                                                                                                                                                       
  File "/var/home/marty/Documents/projects/perun/perun/logic/runner.py", line 27, in <module>                                                                                                                                                                 
    import perun.collect.trace.optimizations.optimization                                                                                                                                                                                                     
  File "/var/home/marty/Documents/projects/perun/perun/collect/trace/optimizations/optimization.py", line 6, in <module>                                                                                                                                      
    from perun.profile.helpers import sanitize_filepart                                                                                                                                                                                                       
ImportError: cannot import name 'sanitize_filepart' from partially initialized module 'perun.profile.helpers' (most likely due to a circular import) (/var/home/marty/Documents/projects/perun/perun/profile/helpers.py)
```

### CODE REVIEW

File: perun/collect/trace/optimizations/optimization.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import collections
 
-from perun.profile.helpers import sanitize_filepart
+from perun.utils.helpers import sanitize_filepart
```
Comment (JiriPavela) [inline perun/collect/trace/optimizations/optimization.py]: Seems like this import could be merged with the one below it, to keep the imports concise.

Comment (martymichal) [reply perun/collect/trace/optimizations/optimization.py]: Good catch!

**Review by JiriPavela — APPROVED** [review]
I fixed the new typing errors that emerged after a new mypy version has been released. I think we're ready to merge now.

**Comment by martymichal** [comment]
I think the last two commits can be squashed.

**Comment by JiriPavela** [comment]
Good idea, I think we could perhaps go even one step further and do squash&merge of the entire PR. What do you think @tfiedor?

**Comment by tfiedor** [comment]
> Good idea, I think we could perhaps go even one step further and do squash&merge of the entire PR. What do you think @tfiedor?

I am not fan of squash, unless the commiter is dirty bastard (i.e., writing commit messages such as `asdf`, or `blabla`, or `MYNAME`). Which is not this case. 

EDIT: Also, you will appreciate more granular history once you bugfix some issue introduced in history, needing `git bisect` and finding precise chunk which caused error. So I would keep the history as it is, and rather focus on having green checkmarks for the PR.

**Comment by martymichal** [comment]
I agree on not squashing the whole PR but the last two commits are related, so I don't see a problem with squashing those two (I regularly amend commits and then force push).

But every maintainer has a different taste, so you decide :).

**Review by tfiedor — APPROVED** [review]

**Pull request merged**